internal/engine/pod.go

```diff
@@ -102,6 +102,11 @@ func ensureManifestTargetWithPod(state *store.EngineState, pod *v1.Pod) (*store.
 	ns := k8s.NamespaceFromPod(pod)
 	hasSynclet := sidecar.PodSpecContainsSynclet(pod.Spec)
 
+	if pod.DeletionTimestamp != nil && !pod.DeletionTimestamp.IsZero() && ms.PodSet.Pods != nil {
+		delete(ms.PodSet.Pods, podID)
```

Comment:
hmmm...will the short-lived pod log stuff still work if we remove the pod immediately?

Reply:
depends on what you mean by "short-lived pod log stuff"

The case covered by the unit test is a Job, where "short-lived" simply means it isn't in state "Running" for very long - it transitions to "Completed" or "Failed", but doesn't actually get deleted - it still shows up in `kubectl get pods`, and doesn't have a `DeletionTimestamp`.

IIUC, the case that might stop working is if a pod gets deleted, we have a chance to grab its logs while it's in the "Terminating" state, but getting logs from a "Terminating" pod is always going to be a race condition, because it might be gone by the time we try to get the logs.